graylog2-server/src/main/java/org/graylog2/rest/RestTools.java

```diff
@@ -144,4 +147,35 @@ public static String getPathFromResource(Resource resource) {
         return path;
 
     }
+
+    public static Response streamResponse(final ProxiedResource.NodeResponse<ResponseBody> nodeResponse,
+                                          final String mediaType,
+                                          final Consumer<Response.ResponseBuilder> additionalResponseBuildingOnSuccess) {
+        if (nodeResponse.isSuccess()) {
+            // we cannot use try-with because the ResponseBody needs to stream the output
+            ResponseBody responseBody = nodeResponse.entity().orElseThrow();
+
+            try {
```

Comment:
I think we can replace the whole `try`-block with a much simpler version:

```suggestion
            final var responseBuilder = Response.ok().type(MediaType.valueOf(mediaType)).entity(responseBody.byteStream());
            if (additionalResponseBuildingOnSuccess != null) {
                additionalResponseBuildingOnSuccess.accept(responseBuilder);
            }
            return responseBuilder.build();
```